services/memory_storage/memory_storage.cpp

```diff
@@ -256,19 +274,82 @@ namespace services {
                   session.data());
             return;
         }
+        auto collection_names = logical_plan->collection_dependencies();
+        context_storage_t collection_dependencies;
```

Comment:
Strange naming:
`auto collection_names = logical_plan->collection_dependencies();` more or less understandable but now I see that `collection_dependencies` might be not the best choice.  
`dependency_tree_collections_names` looks more understandable. 

Also we have
` context_storage_t collection_dependencies;` vs `auto collection_names = logical_plan->collection_dependencies();` - that's mess me up. Let's use 
` context_storage_t collections_context_storage;`